File: src/main/java/de/rubixdev/enchantedshulkers/enchantment/VacuumEnchantment.java
package de.rubixdev.enchantedshulkers.enchantment;

import de.rubixdev.enchantedshulkers.Mod;
import de.rubixdev.enchantedshulkers.config.WorldConfig;
import net.minecraft.item.ItemStack;
import net.minecraft.server.network.ServerPlayerEntity;

public class VacuumEnchantment extends ContainerEnchantment {
    public VacuumEnchantment() {
        super(Mod.PORTABLE_CONTAINER_TARGET);
    }

    @Override
    protected boolean generate() {
        return WorldConfig.generateVacuum();
    }

    public static boolean onItemPickup(ServerPlayerEntity player, ItemStack stack) {
        if (player.isCreative() && !WorldConfig.creativeVacuum()) return false;
        return SiphonEnchantment.onItemPickup(player, stack, Mod.VACUUM_ENCHANTMENT, false, WorldConfig.weakerVacuum());
    }
}

File: src/main/kotlin/de/rubixdev/enchantedshulkers/enchantment/VacuumEnchantment.kt
package de.rubixdev.enchantedshulkers.enchantment

import de.rubixdev.enchantedshulkers.Mod
import de.rubixdev.enchantedshulkers.config.WorldConfig
import net.minecraft.item.ItemStack
import net.minecraft.server.network.ServerPlayerEntity

//#if MC >= 12006
class VacuumEnchantment : ContainerEnchantment(Mod.PORTABLE_CONTAINER_TAG) {
//#else
//$$ class VacuumEnchantment : ContainerEnchantment(Mod.PORTABLE_CONTAINER_TARGET) {
//#endif
    override fun generate() = WorldConfig.generateVacuum

    companion object {
        @JvmStatic
        fun onItemPickup(player: ServerPlayerEntity, stack: ItemStack): Boolean {
            if (player.isCreative && !WorldConfig.creativeVacuum) return false
            return SiphonEnchantment.onItemPickup(player, stack, Mod.VACUUM_ENCHANTMENT, false, WorldConfig.weakerVacuum)
        }
    }
}
